cpp/src/arrow/flight/client.cc

```diff
@@ -963,8 +967,9 @@ class FlightClient::FlightClientImpl {
 
     std::vector<std::unique_ptr<grpc::experimental::ClientInterceptorFactoryInterface>>
         interceptors;
+    middleware = std::move(options.middleware);
```

Comment:
Why this change?

Reply:
Goal was so that middleware was retained in this class so I could push and pop it. I am going to adjust the implementation though.